app/test/export_test.go

```diff
@@ -0,0 +1,56 @@
+package app_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+func TestExportAppStateAndValidators(t *testing.T) {
+	t.Run("should return exported app for version 1", func(t *testing.T) {
+		forZeroHeight := true
+		jailAllowedAddrs := []string{}
+		testApp, _ := SetupTestAppWithUpgradeHeight(t, 3)
+
+		exported, err := testApp.ExportAppStateAndValidators(forZeroHeight, jailAllowedAddrs)
+		require.NoError(t, err)
+		assert.NotNil(t, exported)
+		assert.Equal(t, uint64(1), exported.ConsensusParams.Version.AppVersion)
+	})
+	t.Run("should return exported app for version 2", func(t *testing.T) {
+		forZeroHeight := false
+		jailAllowedAddrs := []string{}
+
+		testApp, _ := SetupTestAppWithUpgradeHeight(t, 3)
+		upgradeToV2(t, testApp)
+
+		exported, err := testApp.ExportAppStateAndValidators(forZeroHeight, jailAllowedAddrs)
+		require.NoError(t, err)
+		assert.NotNil(t, exported)
+		// TODO: the following assertion is commented out because the exported app does not populate consensus params.version
```

Comment:
should we create an issue for this? or it's fine not to support exporting the version?

Reply:
Already an issue https://github.com/celestiaorg/celestia-app/issues/3472

Reply:
Have you tested to make sure that a user can export the state and start up a new node from that exported state i.e. I would assume not saving the consensus params.version would cause issues upon restart

Reply:
I haven't tested but agreed it could cause issues so I expect a user would have to manually populate the app version prior to start.